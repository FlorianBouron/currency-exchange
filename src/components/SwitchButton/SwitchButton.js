import React from "react";
import PropTypes from "prop-types";
import Fab from "@material-ui/core/Fab";
import CompareArrows from "@material-ui/icons/CompareArrows";
import styles from "./SwitchButton.module.scss";

const SwitchButton = ({ className, onClick }) => {
  return (
    <Fab
      aria-label="Switch"
      className={[styles["switch-button-container"], className].join(" ")}
      size="small"
      onClick={onClick}
    >
      <CompareArrows className={styles["switch-button__icon"]} />
    </Fab>
  );
};

SwitchButton.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired
};

SwitchButton.defaultProps = {
  className: null
};

export default SwitchButton;
