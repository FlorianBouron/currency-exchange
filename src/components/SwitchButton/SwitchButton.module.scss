@import "../../constants/colors.scss";

.switch-button-container {
  background-color: $white !important;
  color: $primary-blue !important;
  .switch-button__icon {
    transform: rotate(90deg);
  }
}
