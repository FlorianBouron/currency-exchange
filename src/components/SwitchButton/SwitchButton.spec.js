import React from "React";
import { shallow } from "enzyme";
import SwitchButton from ".";

describe("SwitchButton component", () => {
  const component = shallow(<SwitchButton onClick={() => {}} />);
  it("has working component", () => {
    expect(component).toBeDefined();
  });
  it("expects component to never be changed", () => {
    expect(component).toMatchSnapshot();
  });
});
