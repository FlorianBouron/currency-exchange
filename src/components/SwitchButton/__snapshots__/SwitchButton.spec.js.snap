// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchButton component expects component to never be changed 1`] = `
<WithStyles(ForwardRef(Fab))
  aria-label="Switch"
  className="switch-button-container "
  onClick={[Function]}
  size="small"
>
  <[object Object]
    className="switch-button__icon"
  />
</WithStyles(ForwardRef(Fab))>
`;
